privateca/certificate_authority_basic/main.tf

```diff
@@ -15,52 +15,42 @@
  */
 
 # [START privateca_create_ca]
-resource "google_privateca_certificate_authority" "default" {
+resource "google_privateca_certificate_authority" "root_ca" {
   // This example assumes this pool already exists.
   // Pools cannot be deleted in normal test circumstances, so we depend on static pools
-  pool                     = "my-pool"
-  certificate_authority_id = "my-certificate-authority-hashicorp"
-  location                 = "us-central1"
-  deletion_protection      = false # set to true to prevent destruction of the resource
+  pool                                   = "my-pool"
+  certificate_authority_id               = "my-certificate-authority-root"
+  location                               = "us-central1"
+  deletion_protection                    = false # set to true to prevent destruction of the resource
+  ignore_active_certificates_on_deletion = true
   config {
     subject_config {
       subject {
         organization = "HashiCorp"
         common_name  = "my-certificate-authority"
       }
-      subject_alt_name {
-        dns_names = ["hashicorp.com"]
-      }
     }
     x509_config {
       ca_options {
-        is_ca                  = true
-        max_issuer_path_length = 10
+        # is_ca *MUST* be true for certificate authorities
+        is_ca = true
       }
       key_usage {
         base_key_usage {
-          digital_signature  = true
-          content_commitment = true
-          key_encipherment   = false
-          data_encipherment  = true
-          key_agreement      = true
-          cert_sign          = true
-          crl_sign           = true
-          decipher_only      = true
+          # cert_sign and crl_sign *MUST* be true for certificate authorities
+          cert_sign = true
+          crl_sign  = true
         }
         extended_key_usage {
-          server_auth      = true
-          client_auth      = false
-          email_protection = true
-          code_signing     = true
-          time_stamping    = true
+          server_auth = false
         }
       }
     }
   }
-  lifetime = "86400s"
   key_spec {
     algorithm = "RSA_PKCS1_4096_SHA256"
   }
+  // Root CA should be valid indefinetly e.g. 99 years
```

Comment:
the original configuration here is just matching up with the example used in the main documentation I believe. I think your update here points out something important we should say explicitly even if we leave the configuration to match the existing example with a comment of something like

// lifetime for Root CA should be valid indefinitely so outside of 
// testing environments, update to a longer span of time

Reply:
Fine for me either way. That's the least of my concern with the current documentation, because it's just a setting that is not helpful to run a CA in practice, but otherwise a correct.

Reply:
I don't think we want to leave the impression that root CAs last forever (or 99 years). We want most people to explicitly recognize that root CAs will need to be rotated (whether due to expiry, compromise, or just good hygiene) and plan for that.

Having said that, 86,400 seconds is only 1 day which really doesn't make sense for a root CA.

A more reasonable default lifetime for a root CA is 10 years (or even lower, e.g. 5 years). This is the same default we use for a root CA in gcloud: https://cloud.google.com/sdk/gcloud/reference/privateca/roots/create#--validity

So I'd suggest we use
```
lifetime = "${10 * 365 * 24 * 3600}s"
```

Reply:
done, albeit my personal opinion is slightly different on that topic because if I ran a private key infrastructure I usually move a CA out of service by removing it from the ca store of my devices and by no longer issuing certificates for it. Though someone might also use this product for more distributed use cases with less control. :)
In general trying to set a good example with short lifetimes to train for crypto agility is likely a good thing.